src/components/habitEditor/habitEditorComponents/HabitEditorFrequency.js

```diff
@@ -0,0 +1,118 @@
+import styled from "styled-components";
+
+const HabitEditorFrequency = ({
+	getPrevDate,
+	habitCheckList,
+	getPrevMonth,
+}) => {
+	const dateArr1D = [];
+	const arrayLen = 210;
+	const rowLength = arrayLen / 7;
+	let color = "#ddd";
+	let colorFlag = "g";
+
+	for (let i = 0; i < arrayLen; i++) {
+		habitCheckList.find((dateChecked) => dateChecked === getPrevDate(i))
+			? (colorFlag = "b")
+			: (colorFlag = "g");
+
+		dateArr1D.push(
+			getPrevDate(i).slice(
+				getPrevDate(i).indexOf("/") + 1,
+				getPrevDate(i).lastIndexOf("/")
+			) + colorFlag
+		);
+	}
+
+	const dateArr2D = [];
+	while (dateArr1D.length) dateArr2D.push(dateArr1D.splice(0, rowLength));
+
+	const createTable = () => {
+		let table = [];
+		for (let i = 0; i < 7; i++) {
+			let children = [];
+			for (let j = 0; j < rowLength; j++) {
+				dateArr2D[i][j][dateArr2D[i][j].length - 1] === "b"
+					? (color = "#585FFF")
+					: (color = "#ddd");
+				children.push(<Td theme={{ main: color }}>{}</Td>);
+			}
+			table.push(<Tr>{children}</Tr>);
+		}
+		return table;
+	};
+
+	const monthsArray = [
```

Comment:
This should be stored as constant in another file

Reply:
const monthsArray was not being used so I removed it.